compiler/ballerina-lang/src/main/java/io/ballerina/projects/internal/ManifestBuilder.java

```diff
@@ -767,14 +769,35 @@ private String getStringValueFromDependencyNode(TomlTableNode pkgNode, String ke
 
     private String getStringValueFromPreBuildToolNode(TomlTableNode toolNode, String key, String toolCode) {
         TopLevelNode topLevelNode = toolNode.entries().get(key);
-        String errorMessage = "missing key '[" + key + "]' in table  '[tool." + toolCode +"]' in 'Ballerina.toml'.";
+        String errorMessage = "missing key '[" + key + "]' in table '[tool." + toolCode + "]' in 'Ballerina.toml'.";
         if (topLevelNode == null) {
             reportDiagnostic(toolNode, errorMessage,
-                    ProjectDiagnosticErrorCode.MISSING_TOOL_PROPERTIES_IN_BALLERINA_TOML.diagnosticId(),
+                ProjectDiagnosticErrorCode.MISSING_TOOL_PROPERTIES_IN_BALLERINA_TOML.diagnosticId(),
+                DiagnosticSeverity.ERROR);
+            return null;
+        }
+        ToolNodeValueType toolNodeValueType = getBuildToolTomlValueType(topLevelNode);
+        if (toolNodeValueType.equals(ToolNodeValueType.STRING)) {
```

Comment:
```suggestion
        if (ToolNodeValueType.STRING.equals(toolNodeValueType)) {
```
<constant>.equals(<value>) avoids possible NPE